### DISCUSSION TITLE
Ordering groups in the tablist

### DISCUSSION THREAD

**Discussion by Roy416** [Zrips/CMI#931] [Feature Requests]
**Description of issue or feature request:**

I couldn't find anywhere on how to order players by rank on the tablist. Would be super useful if you could add this, would save us adding another tab list plugin

**Comment by LogGits** [comment]
A feature like this would be awesome! Thanks for the suggestion :+1:

**Comment by Getblock** [comment]
Please implement this... i would hate adding another tab list plugin when cmi already has one.

**Comment by Malachiel87** [comment]
nice suggestion about it!

**Comment by LuisGVE** [comment]
Please implement this

**Comment by OEUG99** [comment]
Please add this!

**Comment by x1p** [comment]
Please please implement this sorting in any form, rank, permission, whatever.

**Comment by RSLuke** [comment]
It would be perfect if we could order players by role weight for luckperms per example or others permissions plugins. It would be more useful than ordering by cmi rank in my situation.

**Comment by FestusDM** [comment]
This is a good feature, definitely. Hope it can be implemented soon.

**Comment by Skyfighter1** [comment]
@Zrips pls add this... so much people want see that feature since 8 month

**Comment by Zrips** [comment]
It is nice feature, but it is one of those more complex ones to handle. Its all based on packets and manipulations on player list to your own liking. I did some experimenting with it, got it working, more or less. Still need to find some time to sit down and finish it up to be in decent condition which will allow sorting and stuff. 
I'm one guy, don't expect from me to add every possible feature, i'm not a superman after all.

**Comment by FestusDM** [comment]
It is a nice feature and yes, understandable with complex tasks to handle. Complete understandable that you are waiting for completion before release. 

If you don't hear it enough, know that we appreciate the work you're doing. Thanks 😄

**Comment by partydev** [comment]
We are also looking forward to this feature. Any updates on this?

**Comment by Skyfighter1** [comment]
Any news?

**Comment by Vehzx** [comment]
Any updates on this? Would love to be able to use this, sorting based off permissions would be great too

**Comment by 4c5458** [comment]
My advice: don't wait for this and look for a real alternative.
TAB reborn is a well maintained tab sorting plugin with all of the requested functionality. This plugin exist for quite a long time and is performing very well and uses very little resources.
CMI is not a "one size fits all" plugin. There are many requests, more than the dev can handle.

**Comment by Vehzx** [comment]
Will look into it, ty

**Comment by R4z0rX** [comment]
I was looking for the same thing and I found this:
**[8.5.4.0 Sort that mess](https://www.spigotmc.org/resources/cmi-280-commands-insane-kits-portals-essentials-economy-mysql-sqlite-much-more.3742/update?update=270435)**

* Added Experimental tablist sorting by multiple criteria.

And then, I found this... 😕
**[8.5.6.0 Changes](https://www.spigotmc.org/resources/cmi-280-commands-insane-kits-portals-essentials-economy-mysql-sqlite-much-more.3742/update?update=272262)**
* For time being completely disabled tablist sorting. Some extra work is needed for it to be viable.

I hope Zrips can go back to this again, and make it work soon...

**Reply by Vehzx** [reply]
Wow this thread is a throwback, I believe I ended up switching tab plugin, I won't post it here as it's irrelevant to this project from Zrips.

My discord is Vehzx#0001 if you want a link to the alternative I used, I believe it's updated for current versions of MC.

**Comment by IronDirt** [comment]
News?

**Comment by mrfloris** [comment]
No news, i know zrips is aware of this highly requested feature. I have no inside info about this though.

**Comment by CloudeLecaw** [comment]
jep this feature is requested and wished since years. im sure its one of the most requested once.

**Reply by RSLuke** [reply]
Agreed

**Comment by Dimaxa131** [comment]
Все еще нету?

**Comment by Chevels** [comment]
Any news for that request ? I'm stuck here in my cmi Tablist config :(

**Comment by ScrapMetalGolem** [comment]
Carrying on the legacy here; this request has survived a global pandemic.

**Comment by CloudeLecaw** [comment]
jup, and its still asked and wanted for by many many cmi users ;)

may one of the most requested features

**Comment by Chevels** [comment]
Since we've been waiting for so long, I think there are 3 solutions:
- either Zrips has nothing to do with this request
- either he coded something in CMI which makes adding this functionality very complicated, so he prefers to turn a deaf ear
- either he is overwhelmed by pending issues (340 !!!) and can no longer manage between the various requests and the correction of problems

Personally I gave up on the idea that he will add this feature x)

**Comment by RSLuke** [comment]
Yup I gave up as well long time ago, I took another plugin for this feature

**Reply by LucasHervieu** [reply]
which plugin did you use?

**Reply by mrfloris** [reply]
> which plugin did you use?

Most use "TAB"

**Reply by ScrapMetalGolem** [reply]
Yeah I'm sure there are several other solutions, but my main use case for CMI is that I can replace the ## other plugins I needed with Essentials all in one package. I'm already using a different solution for chat formatter, and if I need more plugins for stuff like tab, death messages, vanish, spawners, portals, holograms, then CMI becomes less useful. So I tend to use CMI within its capabilities and hope for improvements later, rather than add more solutions to my already boated server :D

**Reply by mrfloris** [reply]
> Yeah I'm sure there are several other solutions, but my main use case for CMI is that I can replace the ## other plugins I needed with Essentials all in one package. I'm already using a different solution for chat formatter, and if I need more plugins for stuff like tab, death messages, vanish, spawners, portals, holograms, then CMI becomes less useful. So I tend to use CMI within its capabilities and hope for improvements later, rather than add more solutions to my already boated server :D

Yep, that's why suggesting it here is great, however, there's no guarantee it will become a feature, and f so, when. It's a highly requested one. Hopefully one day. CMI currently does not sort groups in tablist.yml.

**Comment by CloudeLecaw** [comment]
six years have been passted and its stil one of the most requirements features :-D

well we never give up may seeing this one day :-)

**Comment by mrfloris** [comment]
Tablist.yml has player-group sorting in the latest builds of cmilib and cmi. Upgrade and review tablist.yml's configuration (bottom of page). You can sort ascending or descending by type, and types range from permission node weight to baltop rank, etc.
